README.md
# py_calendars

### This package show the calendar view on the terminal when  you give the year and month also give the satring position of month and year

---

> After install the package import the module 


```
from py_calendars import calendar

calendar(2005,'jan')
```
---
> In terminal you get the out put 

```

      January 2005 

 sun   mon  tus  wed  thr  fri  sat 

                                01
  02   03   04   05   06   07   08
  09   10   11   12   13   14   15
  16   17   18   19   20   21   22
  23   24   25   26   27   28   29
  30   31

```
----
### About  package

>In this package thir are three main part 
1. Finding the given year starting position
2. Finding the given month starting position 
3. use the two resuelt to create the calendar printing design

----
> if you want the year starting position import the year_start ( ) function

```
from py_calendars import year_start

print('ans= ', year_start(2005))
```
#### output
```
ans= 7
```
> That means the 2005 January start with 7 position saturday 
```
      January 2005 

 sun   mon  tus  wed  thr  fri  sat 

                                01
  02   03   04   05   06   07   08
  09   10   11   12   13   14   15
  16   17   18   19   20   21   22
  23   24   25   26   27   28   29
  30   31 
  ```

 >if you want the starting month position use month_start ( ) function

 ```
 from py_calendars import month_start


print('ans= ' month_start(2005,'nov'))
```
#### output
````
ans= 3
````
>that mean the 2005 November start with 3 position Tuesday

```
      November 2005 

 sun   mon  tus  wed  thr  fri  sat 

            01   02   03   04   05
  06   07   08   09   10   11   12
  13   14   15   16   17   18   19
  20   21   22   23   24   25   26
  27   28   29   30

```

CHANGELOG.txt
Change log
==========


0.0.1 (21/10/2021)
------------------
- First Release

testing/tester.py
from calendarpkg.calendar import calen



calen(2021,'mar')

testing/calendarpkg/calendar.py
# to create the calendar design 
from calendarpkg.year_month import year_start

def calen(year,month):
    ans = year_start(year,month)
    if ans ==0:
        ans=7
    # day arrangement 
    mont = {'jan':31,'feb':28,'mar':31,'apr':30,'may':31,'jun':30,'jul':31,'aug':31,'sep':30,'oct':31,'nov':30,'dec':31}
    
    if (year%100)%4 == 0:
        mont['feb']=29
    else:
        mont['feb']=28
    day = mont[month]
    b=ans-1
    d=7-b
    da=1
    for i in range (6):
        if b != 0:
            for i in range(1,b+1):
                print('   ',end='')
                b=0
                
            
        for z in range (d):
            print( '',da,'' , end='')
            da=da+1
            if da > day:
                break
        print()
        if da > day:
                break
        d=7


calendarpkg/calendar.py
# to create the calendar design 
from calendarpkg.year_month import year_start

def calen(year,month):
    ans = year_start(year,month)
    if ans ==0:
        ans=7
    # day arrangement 
    mont = {'jan':31,'feb':28,'mar':31,'apr':30,'may':31,'jun':30,'jul':31,'aug':31,'sep':30,'oct':31,'nov':30,'dec':31}
    
    if (year%100)%4 == 0:
        mont['feb']=29
    else:
        mont['feb']=28
    day = mont[month]
    b=ans-1
    d=7-b
    da=1
    
    m = {'jan':'January','feb':'February','mar':'March','apr':'April','may':'May','jun':'June','jul':'July','aug':'August','sep':'September','oct':'October','nov':'November','dec':'December'}
    print()
    print()
    print('\033[36m','\033[1m','   ',m[month], year,'\033[0m')
    print()
    print('\033[91m','sun','\033[0m','\033[94m','mon  tus  wed  thr  fri  sat','\033[0m')
    print()

    for i in range (6):
        if b != 0:
            for i in range(1,b+1):
                print('   ',' ',end='')
                b=0
                
            
        for z in range (d):
            
                
            print( ' ',"%02d"% da,'', end='')
            
            da=da+1
            if da > day:
                break
        print()
        if da > day:
                break
        d=7
    print()

src/py_calendar/year_month.py
# this is the module to find the every year starting with witch day on jan
# and find the given month starting also


def year_start(year):
    val=0
    if year == 1970: # the year staring with 1970
        return 369%7

    if year < 1970:
        return 'the year equal to 1970 or more than that '
    else:
        n= year-1970
        y=1970
        
        for i in range(n):
           
            if  (y%100)% 4 == 0:
                val=val+366
            elif y == 1970:
                val=val+369 # 370 is the 365 day and extra 4 empty cloumn in the 1970 jan in starting 

            else:
                val=val+365 
            y=y+1
        ans= (val % 7)+1
        return ans
    
def month_start(year,month):   # this section find the month starting day position

        ans = year_start(year)
       
        mm=0
        mont = {'jan':1,'feb':2,'mar':3,'apr':4,'may':5,'jun':6,'jul':7,'aug':8,'sep':9,'oct':10,'nov':11,'dec':12}
        mont_d = {1:31,2:28,3:31, 4:30,5:31,6:30,7:31,8:31,9:30,10:31,11:30,12:31}
        if (year%100)%4 == 0:
            mont_d[2]=29
        else:
            mont_d[2]=28    
        key = int(mont[month])
        
        for z in range(1,key):
            mm = mm + mont_d[z]
            
        mm=mm+ans    
        ans1= mm%7
        if ans1 == 0:
            ans1=7
        return ans1    







src/py_calendar/__init__.py
# to create the calendar design 
from py_calendar.year_month import month_start, year_start


def calendar(year,month):
    ans = month_start(year,month)
    
    # day arrangement 
    mont = {'jan':31,'feb':28,'mar':31,'apr':30,'may':31,'jun':30,'jul':31,'aug':31,'sep':30,'oct':31,'nov':30,'dec':31}
    
    if (year%100)%4 == 0:
        mont['feb']=29
    else:
        mont['feb']=28
    day = mont[month]
    b=ans-1
    d=7-b
    da=1
    
    m = {'jan':'January','feb':'February','mar':'March','apr':'April','may':'May','jun':'June','jul':'July','aug':'August','sep':'September','oct':'October','nov':'November','dec':'December'}
    print()
    print()
    print('\033[36m','\033[1m','   ',m[month], year,'\033[0m')
    print()
    print('\033[91m','sun','\033[0m','\033[94m','mon  tus  wed  thr  fri  sat','\033[0m')
    print()

    for i in range (6):
        if b != 0:
            for i in range(1,b+1):
                print('   ',' ',end='')
                b=0
                
            
        for z in range (d):
            
                
            print( ' ',"%02d"% da,'', end='')
            
            da=da+1
            if da > day:
                break
        print()
        if da > day:
                break
        d=7
    print()
    




